packages/component-base/test/uri-utils.test.js

```diff
@@ -0,0 +1,33 @@
+import { expect } from '@esm-bundle/chai';
+import { arePathsMatching } from '../src/url-utils.js';
+
+describe('url-utils', () => {
+  describe('arePathsMatching', () => {
+    const paths = ['', '/', '/path', 'base/path'];
+
+    it('should return true when paths match', () => {
+      paths.forEach((path) => expect(arePathsMatching(path, path)).to.be.true);
+    });
+
+    it('should return false when paths do not match', () => {
+      expect(arePathsMatching('/path1', '/path2')).to.be.false;
+    });
+
+    it('should return true if only one of the paths has trailing and leading spaces', () => {
+      paths.forEach((path) => {
+        const pathWithExtraSpace = ` ${path} `;
+        expect(arePathsMatching(pathWithExtraSpace, path)).to.be.true;
+        expect(arePathsMatching(path, pathWithExtraSpace)).to.be.true;
+      });
+    });
+
+    it('should return true if only one of the paths has a leading slash', () => {
```

Comment:
```suggestion
    it('should ignore a leading slash in paths', () => {
```

Reply:
Updated the description in the [extracted PR](https://github.com/vaadin/web-components/pull/5997).